commit c1e6fb1aad0d4d4f032d46b6bab093a41c6c82bf
Author: Johannes Berg <johannes@sipsolutions.net>
Date:   Tue Jul 7 03:56:08 2009 +0200

    cfg80211: warn again on spurious deauth
    
    The original code in mac80211 could send a deauth
    frame under certain circumstances even if nothing
    had ever requested an authentication. This has been
    fixed with the rework there, so cfg80211 can now
    warn again about spurious events to catch possible
    future drivers or mac80211 regressions.
    
    Signed-off-by: Johannes Berg <johannes@sipsolutions.net>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/net/wireless/mlme.c b/net/wireless/mlme.c
index f7dc7524e145..960bf60e44e5 100644
--- a/net/wireless/mlme.c
+++ b/net/wireless/mlme.c
@@ -122,15 +122,9 @@ void cfg80211_send_deauth(struct net_device *dev, const u8 *buf, size_t len)
 			break;
 		}
 	}
-/*
- * mac80211 currently triggers this warning,
- * so disable for now (it's harmless, just
- * means that we got a spurious event)
 
 	WARN_ON(!done);
 
- */
-
 	if (wdev->sme_state == CFG80211_SME_CONNECTED) {
 		u16 reason_code;
 		bool from_ap;